nicsdru_nidirect_theme.theme

```diff
@@ -339,6 +339,21 @@ function nicsdru_nidirect_theme_preprocess_page_title(&$variables) {
       }
     }
   }
+
+  // Proprocess titles on certain routes.
+  $route = \Drupal::routeMatch()->getRouteName();
```

Comment:
I'd look at incorporating the previous block of code with this to make it more efficient. So in the switch statement add in a case for 'entity.node.canonical' and then place the **// Preprocess node titles.** code inside that.

Reply:
@omahm - Looking at this again Steve - this is processing the titles on news view pages not nodes.  

I can move it into the switch block in preprocess page function which processes certain routes for view pages - but if you look at https://github.com/dof-dss/nicsdru_nidirect_theme/blob/development/templates/content/page-title.html.twig, it would be normal pattern to use preprocess page title function for setting variables used in that twig. I'm setting {{ title_subtitle }} for that twig in the preprocess page title function.

Reply:
I'm not sure what you mean by moving it to the preprocess_page function?
Example of what I mean is below:
```
/**
 * Implements hook_preprocess_page_title().
 */
function nicsdru_nidirect_theme_preprocess_page_title(&$variables) {

    $route = \Drupal::routeMatch();
    $variables['title_visible'] = TRUE;

    switch ($route->getRouteName()) {
        case "entity.node.canonical" :
            $node = $route->getParameter('node');
            // Landing pages have a boolean field to enable title visibility.
            if ($node->getType() === 'landing_page') {
                // Is the title visible?
                if ($node->field_enable_title->get(0)->value === '0') {
                    $variables['title_visible'] = FALSE;
                }
            }
            break;

        case "nidirect_news.news_listing" :
            $page = \Drupal::request()->get('page');
            if (isset($page) && $page > 0) {
                $variables['title_subtitle'] = t('page ' . ++$page);
            }
            break;

        default:
            break;
    }
}
```

Reply:
Ahhhh right ok!

Reply:
@omahm Done - could you take another look? Ta